packages/core/conductor/src/workflow/workflow.test.ts

```diff
@@ -256,12 +247,13 @@ describe('workflow', () => {
 const createTestExecutionContext = (mocks?: {
   functions?: Context.Tag.Service<FunctionCallService>;
 }): TestEffectLayers => {
-  const logLayer = Layer.succeed(EventLogger, createDxosEventLogger(LogLevel.INFO));
-  const gptLayer = Layer.succeed(GptService, new MockGpt());
-  const spaceService = SpaceService.empty;
-  const queueService = QueueService.notAvailable;
-  const functionCallService = Layer.succeed(FunctionCallService, mocks?.functions ?? FunctionCallService.mock());
-  return Layer.mergeAll(logLayer, gptLayer, spaceService, queueService, FetchHttpClient.layer, functionCallService);
+  return new ServiceContainer()
```

Comment:
can we borrow from effect DI and .provide these?

Reply:
isn't this what its doing, or do you have something else in mind?